code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/impl/Aes128Encrypter.java

```diff
@@ -0,0 +1,75 @@
+package io.cattle.platform.framework.encryption.impl;
+
+import io.cattle.platform.framework.encryption.Encrypter;
+import io.cattle.platform.framework.encryption.EncryptionUtils;
+
+import java.security.Key;
+import java.security.SecureRandom;
+import javax.annotation.PostConstruct;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import javax.inject.Inject;
+
+import org.apache.commons.codec.binary.Hex;
+
+public class Aes128Encrypter extends Encrypter {
+
+    @Inject
+    EncryptionUtils encryptionUtils;
+
+    private String encryptionKey;
+
+    private static final String NAME = "AES128";
+    private Key key;
+    private SecureRandom rn;
+
+    @PostConstruct
+    public void init() {
+        try {
+            rn = new SecureRandom();
+
+            encryptionKey = encryptionUtils.getKeyFromFile(NAME + "Key");
```

Comment:
i don't see that this is hex decoded.  The encryptionUtils should probably get and take a `byte[]` and not have the caller do that.